pkg/ext-proc/backend/pod_reconciler.go

```diff
@@ -0,0 +1,73 @@
+package backend
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"inference.networking.x-k8s.io/gateway-api-inference-extension/api/v1alpha1"
+	logutil "inference.networking.x-k8s.io/gateway-api-inference-extension/pkg/ext-proc/util/logging"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PodReconciler struct {
+	client.Client
+	Datastore *K8sDatastore
+	Scheme    *runtime.Scheme
+	Record    record.EventRecorder
+}
+
+func (c *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
I think a reconciler on pods is not necessary, the pods are already cached in the informer cache, this reconciler seems to just replicate what the informer cache does.

Calling a list using the provided selector should achieve the same effect as getting the list of pods from the datastore.

Reply:
Another issue we will face is missed pod events (e.g., due to delayed InferencePool creation, selector updates, or ext-proc restarts), which will result in stale datastore.